detectable/src/test/java/com/synopsys/integration/detectable/detectables/go/functional/GoModDetectableDepVerifyTest.java

```diff
@@ -0,0 +1,116 @@
+package com.synopsys.integration.detectable.detectables.go.functional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+
+import com.synopsys.integration.bdio.model.Forge;
+import com.synopsys.integration.detectable.Detectable;
+import com.synopsys.integration.detectable.DetectableEnvironment;
+import com.synopsys.integration.detectable.ExecutableTarget;
+import com.synopsys.integration.detectable.detectable.exception.DetectableException;
+import com.synopsys.integration.detectable.detectable.executable.resolver.GoResolver;
+import com.synopsys.integration.detectable.detectables.go.gomod.GoModCliDetectableOptions;
+import com.synopsys.integration.detectable.extraction.Extraction;
+import com.synopsys.integration.detectable.functional.DetectableFunctionalTest;
+import com.synopsys.integration.detectable.util.FunctionalTestFiles;
+import com.synopsys.integration.detectable.util.graph.NameVersionGraphAssert;
+import com.synopsys.integration.executable.ExecutableOutput;
+
+public class GoModDetectableDepVerifyTest extends DetectableFunctionalTest {
+    public GoModDetectableDepVerifyTest() throws IOException {
+        super("gomod");
+    }
+
+    @Override
+    protected void setup() throws IOException {
+        addFile(Paths.get("go.mod"));
+
+        ExecutableOutput goListOutput = new ExecutableOutput(FunctionalTestFiles.asString("/go/go-list.xout"), StringUtils.EMPTY);
+        addExecutableOutput(goListOutput, "go", "list", "-m", "-json");
+
+        ExecutableOutput goVersionOutput = createStandardOutput(
+            "go version go1.16.5 darwin/amd64"
+        );
+        addExecutableOutput(goVersionOutput, "go", "version");
+
+        ExecutableOutput goListUJsonOutput = new ExecutableOutput(FunctionalTestFiles.asString("/go/go-list-all.xout"), StringUtils.EMPTY);
```

Comment:
@taikuukaits do you have any qualms with me adding a `DetectableFunctionalTest:: createStandardOutputFromResource` method? Or are we trying to discourage the use of resource files?

Reply:
I think as long as FunctionalTestFiles.asString is permitted I would begrudgingly permit createStandardOutputFromResource. Though I still do think large inputs must be defended. I feel in this case go-list-all makes sense as a file, but I don't see why go-list is a file. (My guess is because they are shared between two tests?)